e2e-polybft/framework/test-cluster.go

```diff
@@ -456,6 +458,12 @@ func WithRewardWallet(rewardWallet string) ClusterOption {
 	}
 }
 
+func WithPredeploy() ClusterOption {
+	return func(h *TestClusterConfig) {
+		h.PredeployNonNative = true
```

Comment:
I am not a fan of this approach. What I would do is rename this `PredeployNonNative` to `PredeployContract` which would be a string `<address:artifactName>`, or a small struct like:

```go
    type predeployInfo struct {
         address types.Address
         artifactName string
    }

```

That way, we can use this to pre-deploy whatever we want in tests, and not just some fixed address and fixed artifact.

Reply:
Reminder to address this comment @dusannosovic-ethernal